src/views/AddItem.jsx

```diff
@@ -1,22 +1,33 @@
 import { useState } from 'react';
 import { addItem } from '../api/firebase';
 
-export function AddItem({ listToken }) {
+export function AddItem({ listToken, data, setData }) {
 	const [daysUntilNextPurchase, setTimeFrame] = useState('7');
 	const [itemName, setItem] = useState('');
-	const [itemData, setData] = useState({});
 	const [error, setError] = useState(false);
+	const [duplicateError, setDuplicateError] = useState(false);
 	const [success, setSuccess] = useState(false);
 
+	// check for duplicate item in list
+	const isDuplicate = (data) =>
+		data.name.toLowerCase().replace(/ /g, '') ===
```

Comment:
I would try to stay away from 'data' - someone who didn't write the code won't understand what it represents on first glance, I think from context since you're accessing the name it's the listItem but that's on line 2 -- you want to try to give that context the very first time you reference something.

Reply:
^ yes please for some context, I'm guessing this is checking to see whether something exists already in the database, but I wasn't sure at first!

Reply:
I went ahead and renamed the data prop to itemList. Hopefully that provides some more clarity since data is the list of items.